src/efi.rs

```diff
@@ -860,4 +1103,170 @@ Boot0003* test";
         );
         Ok(())
     }
+
+    #[test]
+    fn test_extend_payload() -> Result<()> {
+        use std::fs;
+        use tempfile::TempDir;
+
+        let temp_sysroot = TempDir::new()?;
+        let temp_src = TempDir::new()?;
+
+        let sysroot_path = temp_sysroot.path().to_str().unwrap();
+        let src_path = temp_src.path().to_str().unwrap();
+
+        // mockup data source: /usr/share/uboot/rpi/
+        // content: u-boot.bin, overlays/i2c.dtb
+        // mockup rpm: uboot-images-2023.04-2.fc42.noarch
+        // mockup rpm_db: /usr/lib/sysimage/rpm/Packages
+        let src_uboot_dir = temp_src
+            .path()
+            .join("usr")
+            .join("share")
+            .join("uboot")
+            .join("rpi");
+        fs::create_dir_all(&src_uboot_dir)?;
+
+        let src_overlays_dir = src_uboot_dir.join("overlays");
+        fs::create_dir_all(&src_overlays_dir)?;
+
+        // Create content files
+        let uboot_bin = src_uboot_dir.join("u-boot.bin");
+        fs::write(&uboot_bin, b"uboot binary content")?;
+        let overlay_dtb = src_overlays_dir.join("i2c.dtb");
+        fs::write(&overlay_dtb, b"device tree overlay content")?;
+
+        // Create a mockup RPM database structure
+        let rpm_db_dir = temp_sysroot
+            .path()
+            .join("usr")
+            .join("lib")
+            .join("sysimage")
+            .join("rpm");
+        fs::create_dir_all(&rpm_db_dir)?;
+        fs::write(rpm_db_dir.join("Packages"), b"fake rpm database file")?;
+
+        // Create a mock rpm script that returns uboot-images package data
+        let mock_rpm_dir = TempDir::new()?;
+        let mock_rpm_script = mock_rpm_dir.path().join("rpm");
+
+        let mock_script_content = r#"#!/bin/bash
+# Mock rpm script for testing
+if [[ "$*" == *"-q"* ]] && [[ "$*" == *"-f"* ]]; then
+    # Return mock uboot-images package data in the expected format: nevra,buildtime
+    echo "uboot-images-2023.04-2.fc42.noarch,1681234567"
+    exit 0
+fi
+# For any other rpm command, just fail
+exit 1
+"#
+        .to_string();
+
+        fs::write(&mock_rpm_script, mock_script_content)?;
+
+        #[cfg(unix)]
+        {
+            use std::os::unix::fs::PermissionsExt;
+            let mut perms = fs::metadata(&mock_rpm_script)?.permissions();
+            perms.set_mode(0o755);
+            fs::set_permissions(&mock_rpm_script, perms)?;
+        }
+
+        let original_path = std::env::var("PATH").unwrap_or_default();
+        let new_path = format!("{}:{}", mock_rpm_dir.path().display(), original_path);
+        std::env::set_var("PATH", &new_path);
```

Comment:
This is fully `unsafe` for good reason in Rust 2024: https://doc.rust-lang.org/edition-guide/rust-2024/newly-unsafe-functions.html#stdenvset_var-remove_var

It's more painful but basically if we wanted to mock RPM we'd have to explicitly pass it as a parameter.